pennylane_qiskit/vqe_runtime_runner.py

```diff
@@ -0,0 +1,488 @@
+        project (str): Ibmq provider project.
+
+    Returns:
+        str: Program id that can be used to run the program.
+    """
+
+    connect(kwargs)
+
+    # Specify a single hub, group and project
+    hub = kwargs.get("hub", "ibm-q")
+    group = kwargs.get("group", "open")
+    project = kwargs.get("project", "main")
+
+    meta = {
+        "name": "vqe-runtime",
+        "description": "A sample VQE program.",
+        "max_execution_time": 100000,
+        "spec": {},
+    }
+
+    meta["spec"]["parameters"] = {
+        "$schema": "https://json-schema.org/draft/2019-09/schema",
+        "properties": {
+            "hamiltonian": {
+                "description": "Hamiltonian whose ground state we want to find.",
+                "type": "array",
+            },
+            "x0": {
+                "description": "Initial vector of parameters for the quantum circuit.",
+                "type": "array",
+            },
+            "ansatz": {
+                "description": "Qiskit circuit or name of ansatz quantum circuit to use, default='EfficientSU2'",
+                "type": "[QuantumCircuit,string]",
+                "default": "EfficientSU2",
+            },
+            "ansatz_config": {
+                "description": "Configuration parameters for the ansatz circuit.",
+                "type": "dict",
+            },
+            "optimizer": {
+                "description": "Classical optimizer to use, default='SPSA'.",
+                "type": "string",
+                "default": "SPSA",
+            },
+            "optimizer_config": {
+                "description": "Configuration parameters for the optimizer.",
+                "type": "dict",
+            },
+            "shots": {
+                "description": "The number of shots used for each circuit evaluation.",
+                "type": "integer",
+            },
+            "use_measurement_mitigation": {
+                "description": "Use measurement mitigation, default=False.",
+                "type": "boolean",
+                "default": False,
+            },
+        },
+        "required": ["hamiltonian", "x0"],
+    }
+
+    meta["spec"]["return_values"] = {
+        "$schema": "https://json-schema.org/draft/2019-09/schema",
+        "description": "Final result in SciPy optimizer format",
+        "type": "object",
+    }
+
+    meta["spec"]["intermadiate_results"] = {
+        "$schema": "https://json-schema.org/draft/2019-09/schema",
+        "description": "Dictionnary containing: "
+        "The number of evaluation at current optimization step."
+        "Parameter vector at current optimization step."
+        "Function value at the current optimization step."
+        "The size of the step.",
+        "type": "dict",
+    }
+
+    provider = IBMQ.get_provider(hub=hub, group=group, project=project)
+
+    program_id = provider.runtime.upload_program(
+        data="runtime_programs/vqe_runtime_program.py", metadata=meta
+    )
```

Comment:
For some reason, my computer does not detect this file with the relative path. I have had to set the absolute path to be able to test it 🤔

Reply:
It is strange I will verify it

Reply:
It works on my side, what about you @antalszava ?

Reply:
Yep, also getting this error:
```python
~/anaconda3/lib/python3.8/site-packages/qiskit/providers/ibmq/runtime/ibm_runtime_service.py in upload_program(self, data, metadata)
    355         if "def main(" not in data:
    356             # This is the program file
--> 357             with open(data, "r") as file:
    358                 data = file.read()
    359 

FileNotFoundError: [Errno 2] No such file or directory: 'runtime_programs/vqe_runtime_program.py'
```
@rmoyard it might be working for you as you're trying it from the `pennylane_qiskit` working directory? If so, `runtime_programs` is found in the current working directory, but otherwise there might be no way of finding it.

Likely the path to `runtime_programs/vqe_runtime_program.py` has to be put in the path so that it's to be found.

One thing that comes to mind is using a well-defined location by using [appdirs](https://pypi.org/project/appdirs/).

Also, the `runtime_programs/vqe_runtime_program.py` path might not work on Windows, using `pathlib` or `os.path` would potentially be required.

Reply:
Okay thank you @antalszava  I will make the necessary changes!